precompiles/xcm/src/lib.rs

```diff
@@ -51,11 +52,24 @@ where
 
         match selector {
             // Dispatchables
-            Action::AssetsWithdraw => Self::assets_withdraw(handle),
+            Action::AssetsWithdrawNative => {
+                Self::assets_withdraw(handle, BeneficiaryType::AccountId32)
+            }
+            Action::AssetsWithdrawEvm => {
+                Self::assets_withdraw(handle, BeneficiaryType::AccountKey20)
+            }
         }
     }
 }
 
+/// The supported beneficiary account types
+enum BeneficiaryType {
+    /// 256 bit (32 byte) public key
+    AccountId32,
+    /// 160 bit (20 byte) address is expected
+    AccountKey20,
```

Comment:
Technically it’s not public _key_, it’s id.

Reply:
I know, hence the comment above.

However, I used `AccountKey20` to keep it consistent with `Junction` enum.

Reply:
So, what’s about to drop Id/Kay from name, like “Account32 | Account20”?

Reply:
Sure, let's do that.